File: py/other/JosephusRing.py
#!/usr/bin/env python
# -*- coding:gbk -*-

class JosephusRing:
    def solution(self, num: int, interval: int) -> int:
        result = 0
        for i in range(1, num):
            result = (result + interval) % (i + 1)
        return result

File: cpp/other/JosephusRing.cpp
#include <iostream>
using namespace std;

class JosephusRing {
public:
    int solution(int num, int interval) {
        int result = 0;
        for (int i = 1; i < num; ++i) {
            result = (result + interval) % (i + 1);
        }
        return result;
    }
};
